ntp-proto/src/nts_record.rs

```diff
@@ -416,10 +421,37 @@ pub enum KeyExchangeError {
     IncompleteResponse,
 }
 
+/// From https://www.iana.org/assignments/aead-parameters/aead-parameters.xhtml
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Default)]
+#[repr(u8)]
+enum AeadAlgorithm {
+    #[default]
+    AeadAesSivCmac256 = 15,
+    AeadAesSivCmac512 = 17,
+}
+
+impl AeadAlgorithm {
+    // per https://www.rfc-editor.org/rfc/rfc8915.html#section-5.1
+    pub const fn c2s_context(self) -> [u8; 5] {
+        // The final octet SHALL be 0x00 for the C2S key
+        [0, 0, 0, self as u8, 0]
+    }
+
+    // per https://www.rfc-editor.org/rfc/rfc8915.html#section-5.1
+    pub const fn s2c_context(self) -> [u8; 5] {
+        // The final octet SHALL be 0x01 for the S2C key
+        [0, 0, 0, self as u8, 1]
+    }
+
+    const IN_ORDER_OF_PRECEDENCE: &'static [Self] =
```

Comment:
Nit: I think you mean preference, not precedence.

Reply:
yup